src/test/jenkins/jenkinsfiles/Environment_Jenkinsfile

```diff
@@ -1,18 +1,24 @@
-pipeline {
-    agent any
-    environment {
-        LEVAR1 = "LE VALUE 1"
-    }
-    stages {
-        stage('Example') {
-            environment {
-                LEVAR1 = "LE NEW VALUE"
-                LEVAR2 = "A COPY OF ${env.LEVAR1} in build#${currentBuild.number}"
-            }
-            steps {
-                echo "LEVAR1 ${env.LEVAR1}"
-                echo "LEVAR2 ${env.LEVAR2}"
+def call() {
```

Comment:
could you please add a comment why to wrap pipeline into a call method as it is not so obvious
If it is an additional testcase then create a new Jenkinsfile

Reply:
We wrap the entire pipeline script in a method in our project to be able to access it from a jenkins pipelinescript job.

In the jenkins job we don't have to have the scm url in each build. We just have 1 library which we can use var groovyscripts from.

Inspiration taken from:
https://www.jenkins.io/doc/book/pipeline/shared-libraries/

So basically we have a method we can call to build a pipeline block. We do some initialisation in the block above pipeline{

This way we can use custom labels depending on env or input params. This allows us to reuse pipelines over a multitude of jobs.

Reply:
The call() at the bottom isn't required in jenkins, but i wanted to avoid going loadScript("scriptname")()

Reply:
I re-read the request and made a new testcase to show the scoping works.